App/Components/Common/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import DateTimePicker from 'react-native-modal-datetime-picker';
+import PropTypes from 'prop-types';
+import { TouchableOpacity } from 'react-native';
+
+// Common
+import Text from '../Text';
+
+class TimePicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDateTimePickerVisible: false,
+      date: props.time || '--:--',
+    };
+  }
+
+  showDateTimePicker = () => {
+    this.setState({ isDateTimePickerVisible: true });
+  };
+
+  hideDateTimePicker = () => {
+    this.setState({ isDateTimePickerVisible: false });
+  };
+
+  handleDatePicked = date => {
+    this.setState({ date: `${date.getHours()}:${date.getMinutes()}` });
+    this.hideDateTimePicker();

```

Comment:
Es mejor que llames la funcion del hideDateTimePicker ddespues en el callback del setState, para asegurarse de que el state se setio, this.setState({}, callback)